src/axom/quest/detail/MarchingCubesImpl.hpp

```diff
@@ -0,0 +1,997 @@
+// Copyright (c) 2017-2024, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#include "axom/config.hpp"
+
+// Implementation requires Conduit.
+#ifndef AXOM_USE_CONDUIT
+  #error "MarchingCubesImpl.hpp requires conduit"
+#endif
+#include "conduit_blueprint.hpp"
+
+#include "axom/core/execution/execution_space.hpp"
+#include "axom/slic/interface/slic_macros.hpp"
+#include "axom/quest/ArrayIndexer.hpp"
+#include "axom/quest/MeshViewUtil.hpp"
+#include "axom/quest/detail/MarchingCubesSingleDomain.hpp"
+#include "axom/primal/geometry/Point.hpp"
+#include "axom/primal/constants.hpp"
+#include "axom/mint/execution/internal/structured_exec.hpp"
+#include "axom/fmt.hpp"
+
+namespace axom
+{
+namespace quest
+{
+namespace detail
+{
+namespace marching_cubes
+{
+/*!
+  @brief Computations for MarchingCubesSingleDomain
+
+  Spatial dimension templating is here, to keep out of higher level
+  classes MarchCubes and MarchingCubesSingleDomain.
+
+  ExecSpace is the general execution space, like axom::SEQ_EXEC and
+  axom::CUDA_EXEC<256>.
+
+  See MarchingCubesImpl for the difference between that class and
+  MarchingCubesImpl.
```

Comment:
This sentence is cryptic: please explain what "that class" means.  Perhaps refer to a file name.

Reply:
Shoot.  Another comment that I should have removed when recombining the full-parallel and hybrid-parallel classes.